test/test_other.py

```diff
@@ -3568,6 +3568,23 @@ def test_js_lib_no_override(self):
     err = self.expect_fail([EMCC, 'duplicated_func.c'] + self.get_emcc_args())
     self.assertContained('error: Symbol re-definition in JavaScript library: duplicatedFunc. Do not use noOverride if this is intended', err)
 
+
+  def test_override_stub(self):
+    create_file('override_fork.c', '''
+      int fork() {
+        return 0;
+      }
+
+      int main() {
+        fork();
+        return 0;
+      }
+    ''')
+
+    proc = self.run_process([EMCC, 'override_fork.c', '-sALLOW_UNIMPLEMENTED_SYSCALLS=1'], stderr=PIPE)
```

Comment:
You can/should use `do_runf('override_fork.c', <expected_output>)` here I think.

Also, can you remove the `ALLOW_UNIMPLEMENTED_SYSCALLS` setting here?

Reply:
I can, but we just test whether it compiles, no?
This setting has no effect.

Reply:
I think it this test should both compile and run, no?

Reply:
To test if `weak` works, we just need to compile and link, but we can run too =)

Right now it works even before the PR ;(

Reply:
I think it will fail if you reference a second symbol from the same file (see my comment above).

I'd like to see it not just link but run, to prove that the correct symbol is actually used at runtime.

Reply:
Ok, changed. Now it fails even with weak.

Reply:
Did you remember to rebuild the stub library?

Reply:
I executed `emcc --clear-cache`

Reply:
```bash
➜  emscripten git:(weak-stubs) emcc --clear-cache
emcc:INFO: clearing cache as requested by --clear-cache: `/opt/projects/emsdk/upstream/emscripten/cache`
shared:INFO: (Emscripten: Running sanity checks)
➜  emscripten git:(weak-stubs) ./test/runner other.test_override_stub
posixtestsuite not found (run git submodule update --init?)
Test suites:
['test_other']
Running test_other: (1 tests)
(checking sanity from test runner)
shared:INFO: (Emscripten: Running sanity checks)
test_override_stub (test_other.other) ... Clearing existing test directory
wasm-ld: error: duplicate symbol: fork
>>> defined in /var/folders/8c/z0mt42s97yz4ts2sxkbt_zv40000gn/T/emscripten_temp_uskclfoi/test_override_stub_0.o
>>> defined in /opt/projects/emscripten/cache/sysroot/lib/wasm32-emscripten/libstubs-debug.a(emscripten_libc_stubs.o)
emcc: error: '/opt/projects/emsdk/upstream/bin/wasm-ld -o test_override_stub.wasm /var/folders/8c/z0mt42s97yz4ts2sxkbt_zv40000gn/T/emscripten_temp_uskclfoi/test_override_stub_0.o -L/opt/projects/emscripten/cache/sysroot/lib/wasm32-emscripten -lGL -lal -lhtml5 -lstubs-debug -lnoexit -lc-debug -ldlmalloc -lcompiler_rt -lsockets -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr --allow-undefined-file=/var/folders/8c/z0mt42s97yz4ts2sxkbt_zv40000gn/T/tmpg42q57dd.undefined --strip-debug --export-if-defined=main --export-if-defined=__start_em_asm --export-if-defined=__stop_em_asm --export-if-defined=__start_em_lib_deps --export-if-defined=__stop_em_lib_deps --export-if-defined=__start_em_js --export-if-defined=__stop_em_js --export-if-defined=__main_argc_argv --export-if-defined=fflush --export=emscripten_stack_get_end --export=emscripten_stack_get_free --export=emscripten_stack_get_base --export=emscripten_stack_get_current --export=emscripten_stack_init --export=stackSave --export=stackRestore --export=stackAlloc --export=__errno_location --export=__get_temp_ret --export=__set_temp_ret --export=__wasm_call_ctors --export-table -z stack-size=65536 --initial-memory=16777216 --no-entry --max-memory=16777216 --stack-first' failed (returned 1)
```